### PR TITLE
Integration with Redis to optionally replace MongoDB for Query Metadata Store 

### PR DESCRIPTION
**Pull request by aterleto**
@pierredebelen

What type of PR is this?

    Improvement

What does this PR do / why is it needed ?

Allows Redis to be optionally used instead of MongoDB for the Query Metadata Store
Which issue(s) this PR fixes:

N/A

Fixes #
Other notes for reviewers:
Does this PR introduce a user-facing change?

### CODE REVIEW

File: legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/ApplicationQuery.java [hunk 1]
```diff
@@ -50,9 +54,16 @@ public class ApplicationQuery
 {
     private final QueryStoreManager queryStoreManager;
 
-    public ApplicationQuery(MongoClient mongoClient)
+    public ApplicationQuery(Object client)
```
Comment (rafaelbey) [inline legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/ApplicationQuery.java:57]: Lets avoid using Object, and either create two constructors, or make the input parameter to QueryStoreManager

File: legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/configuration/ApplicationQueryConfiguration.java [hunk 2]
```diff
@@ -17,10 +17,25 @@
...
 import com.mongodb.client.MongoClients;
 import org.finos.legend.engine.shared.core.vault.Vault;
+import redis.clients.jedis.DefaultJedisClientConfig;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisPooled;
 
 public class ApplicationQueryConfiguration
 {
-    public static MongoClient getMongoClient()
+    public static Object getQueryStoreClient()
```
Comment (rafaelbey) [inline legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/configuration/ApplicationQueryConfiguration.java:26]: Lets avoid Object, and return QueryStoreManager.

**Review by rafaelbey — CHANGES_REQUESTED** [review]
Using Object should be last resort, and usually there are alternatives to make the code more readable and robust.

File: legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/configuration/ApplicationQueryConfiguration.java (hunk 2 above)
Comment (aterleto) [reply legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/configuration/ApplicationQueryConfiguration.java]: Hi @rafaelbey, I removed the use of Object but it required me to push the conditional configuration logic, which decides between Redis and MongoDB, to the Server layer. As I did that I noticed I missed committing Redis client dependencies so those are now included to. This is ready for your review. Thanks, Allen

**Comment by finos-admin** [comment]
This PR is stale because it has been open for 30 days with no activity. Please remove stale label or add any comment to keep this open. Otherwise this will be closed in 5 days.

**Comment by finos-admin** [comment]
This PR was closed because it has been inactive for 35 days. Please re-open if this PR is still relevant.

**Pull request closed without merging**